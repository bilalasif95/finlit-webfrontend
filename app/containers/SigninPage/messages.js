/*
 * Signin Page Messages
 *
 * This contains all the text for the Signin Page component.
 */
import { defineMessages } from 'react-intl';

export const scope = 'boilerplate.containers.SigninPage';

export default defineMessages({
  WelcomeBack: {
    id: `${scope}.WelcomeBack`,
    defaultMessage: 'Welcome to FinLit',
  },
  EmailAddress: {
    id: `${scope}.EmailAddress`,
    defaultMessage: 'Email Address',
  },
  Password: {
    id: `${scope}.Password`,
    defaultMessage: `Password`,
  },
  RememberMe: {
    id: `${scope}.RememberMe`,
    defaultMessage: 'Remember me',
  },
  ForgotPassword: {
    id: `${scope}.ForgotPassword`,
    defaultMessage: `Forgot Password?`,
  },
  Login: {
    id: `${scope}.Login`,
    defaultMessage: 'Log In',
  },
  DontHaveAccount: {
    id: `${scope}.DontHaveAccount`,
    defaultMessage: `Don't have account?`,
  },
  CreateAnAccount: {
    id: `${scope}.CreateAnAccount`,
    defaultMessage: 'Sign Up',
  },
  WellDone: {
    id: `${scope}.WellDone`,
    defaultMessage: 'Well done! Your email has been verified.',
  },
});
