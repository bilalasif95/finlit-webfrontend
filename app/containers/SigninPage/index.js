/*
 * Sign In Page
 */
import React, { useState } from 'react';
import { Helmet } from 'react-helmet';
import { FormattedMessage } from 'react-intl';
import { Button, FormGroup, Label, Input } from 'reactstrap';
import history from 'utils/history';
import { Link, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import UseEnterKeyListener from "../../config/useEnterKeyListener";

import axios from 'axios';
import { ToastContainer, toast } from 'react-toastify';
import {
  mapStateToProps,
  mapDispatchToProps,
} from '../reduxSetup/actions/registeration';
import messages from './messages';
import 'react-toastify/dist/ReactToastify.css';
import { API } from '../../config/config';
// import { response } from 'express';
const SigninPage = props => {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [error, setError] = useState('');
  const [btnClick, setBtnClick] = useState(false);
  const [isDisabled, setDisabled] = useState(false);
  UseEnterKeyListener({
    querySelectorToExecuteClick: "#submitButton"
  });
  const [rememberMe, setRememberMe] = useState(false);

  React.useEffect(() => {
    setEmail(localStorage.getItem('remember_me_email') || '');
    setPassword(localStorage.getItem('remember_me_password') || '');
    if (localStorage.getItem('remember_me_email') && localStorage.getItem('remember_me_password')) {
      setRememberMe(true);
    }
  }, [])
  const login = () => {
    setError('');
    debugger;
    if (!email) {
      setError('Please enter email');
      return;
    }
    if (!password) {
      setError('Please enter Password');
      return;
    }
    if (rememberMe) {
      localStorage.setItem('remember_me_email', email);
      localStorage.setItem('remember_me_password', password);
    }
    setBtnClick(true);
    axios
      .post(`${API}api/auth/login`, { email, password })
      .then(res => {
        if (res.data.status === 200) {
          localStorage.setItem('token', res.data.data.accessToken);
          localStorage.setItem(
            'userInfo',
            JSON.stringify(res.data.data.user && res.data.data.user),
          );
          props.Login(res.data.data.user);
          history.push('/');
        } else {
          localStorage.setItem('email', email);
          history.push('/two_fa');
        }
      })
      .catch(err => {
        toast.error(
          err.response && err.response.data.message
            ? err.response.data.message.toString()
            : 'Message Not Readable',
        );
        setTimeout(() => {
          setBtnClick(false);
        }, 5000);
      });
  };
  return (
    <>
      <div className="registration_page">
        <Helmet>
          <title>- Sign In</title>
          <meta name="description" content="FinLit - Sign in Page" />
        </Helmet>
        <div className="form_container Login-form">
          <div className="form_content">
            <h2>
              <FormattedMessage {...messages.WelcomeBack} />
            </h2>
            <div className="form">
              <FormGroup>
                <Label for="email">
                  <FormattedMessage {...messages.EmailAddress} />
                </Label>
                <Input
                  type="email"
                  name="email"
                  id="email"
                  defaultValue={email}
                  onChange={e => setEmail(e.target.value)}
                  placeholder="Your email"
                />
              </FormGroup>
              <FormGroup className="form_err">
                <Label for="password">
                  <FormattedMessage {...messages.Password} />
                </Label>
                <Input
                  type="password"
                  name="password"
                  id="password"
                  defaultValue={password}
                  onChange={e => setPassword(e.target.value)}
                  onKeyDown = {(e) => {
                    if (e.code === "Enter" || e.code === "NumpadEnter") {
                      console.log("Enter key was pressed. Run your function.");
                      e.preventDefault();
                      login()
                    }
                  }}
                  placeholder="******"
                />
                <div className="error-box">
                  {error && <p className="error">{error}</p>}
                </div>
              </FormGroup>
              <div className="remember_forgot">
                <FormGroup check>
                  <Label check className="check-box">
                    <Input defaultValue={rememberMe} value={rememberMe} type="checkbox" onChange={(e) => {
                      setRememberMe(e.target.value);
                    }} />
                    <FormattedMessage {...messages.RememberMe} />
                  </Label>
                </FormGroup>
                <Link to="/forgot_password">
                  <FormattedMessage {...messages.ForgotPassword} />
                </Link>
              </div>
              <Button id="submitButton" onClick={login} disabled={btnClick || !email || !password}>
                <FormattedMessage {...messages.Login} />
              </Button>
              <div className="reg_footer">
                <p>
                  <FormattedMessage {...messages.DontHaveAccount} />
                  &nbsp;
                  <Link to="/signup">
                    <FormattedMessage {...messages.CreateAnAccount} />
                  </Link>
                </p>
              </div>
            </div>
          </div>
        </div>
        <div className="img_container" />
      </div>
      <ToastContainer />
    </>
  );
};

export default withRouter(
  connect(mapStateToProps, mapDispatchToProps)(SigninPage)
);
