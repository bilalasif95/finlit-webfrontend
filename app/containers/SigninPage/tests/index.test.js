import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
// import { BrowserRouter } from 'react-router-dom';
import { Router } from 'react-router-dom';
import history from 'utils/history';
import Store2 from 'containers/reduxSetup/store';
import SigninPage from '../index';
describe('SigninPage />', () => {
  let store;
  beforeAll(() => {
    // store = configureStore({}, browserHistory);
    store = Store2;
  });
  it('should render its heading', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale="en">
          {/* <BrowserRouter> */}
          <Router history={history}>
            <SigninPage />
          </Router>
          {/* </BrowserRouter> */}
        </IntlProvider>
      </Provider>,
    );

    expect(firstChild).toMatchSnapshot();
  });
});
