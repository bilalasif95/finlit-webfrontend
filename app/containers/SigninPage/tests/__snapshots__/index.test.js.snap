// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SigninPage /> should render its heading 1`] = `
<div
  class="registration_page"
>
  <div
    class="form_container"
  >
    <div
      class="form_content"
    >
      <h2>
        <span>
          Welcome Back
        </span>
      </h2>
      <div
        class="form"
      >
        <div
          class="form-group"
        >
          <label
            class=""
            for="email"
          >
            <span>
              Email Address
            </span>
          </label>
          <input
            class="form-control"
            id="email"
            name="email"
            placeholder="Your email"
            type="email"
          />
        </div>
        <div
          class="form-group"
        >
          <label
            class=""
            for="password"
          >
            <span>
              Password
            </span>
          </label>
          <input
            class="form-control"
            id="password"
            name="password"
            placeholder="******"
            type="password"
          />
        </div>
        <div
          class="remember_forgot"
        >
          <div
            class="form-check"
          >
            <label
              class="form-check-label"
            >
              <input
                class="form-check-input"
                type="checkbox"
              />
              <span>
                Remember me
              </span>
            </label>
          </div>
          <a
            href="/"
          >
            <span>
              Forgot Password?
            </span>
          </a>
        </div>
        
        <button
          class="btn btn-secondary"
          type="button"
        >
          <span>
            Log in
          </span>
        </button>
        <div
          class="reg_footer"
        >
          <p>
            <span>
              Don't have account?
            </span>
            Â 
            <a
              href="/signup"
            >
              <span>
                Create an account
              </span>
            </a>
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    class="img_container"
  />
</div>
`;
